### ISSUE TITLE
Poor results for Tel Aviv and Jerusalem with search endpoint

### ISSUE THREAD

**Issue by rmglennon** [pelias/pelias#648]
#### Here's what I did :innocent:
This came in through support (see Desk 888).

When searching for Tel Aviv or Jerusalem with wof and locality, search returns nothing or poor results. Autocomplete, however, is pretty good.

Examples:
https://mapzen.com/search/explorer/?query=search&text=tel%20aviv&layers=locality
https://mapzen.com/search/explorer/?query=search&text=jerusalem&layers=locality

---
#### Here's what I got :scream_cat:
Tel Aviv returns nothing.

Jerusalem puts Australia first, then Palestine...

 Jerusalem, Australia
 Jerusalem, Palestine
 Jerusalem, OH, USA
 Jerusalém, Brazil
 Jerusalém, Brazil

The Jerusalem, Palestine entry is this:

"properties": {
        "id": "421168681",
        "gid": "whosonfirst:locality:421168681",
        "layer": "locality",
        "source": "whosonfirst",
        "source_id": "421168681",

However, a Place search for this ID returns a label of "Jerusalem, Israel", so something is going on there, too.

https://mapzen.com/search/explorer/?query=place&ids=whosonfirst%3Alocality%3A421168681

Spelunker record: https://whosonfirst.mapzen.com/spelunker/id/421168681/

---
#### Here's what I was expecting :sparkles:
Expecting search results to be better, or similar to what you get with Autocomplete.

https://mapzen.com/search/explorer/?query=autocomplete&text=tel%20aviv&sources=wof&layers=locality
Tel Aviv-Yafo, Israel

https://mapzen.com/search/explorer/?query=autocomplete&text=jerusalem&layers=locality

Jerusalem, Israel
 Town of Jerusalem, NY, USA
 New Jerusalem, PA, USA
 Jerusalem, OH, USA
 Jerusalem, OH, USA
 Jerusalem, Mexico
 Sítio Jerusalém, Brazil
 Jerusalem, GA, USA
 Jerusalem (Al Quds), Israel
 Klein Jerusalem, Germany

---
#### Here's what I think could be improved :trophy:
See what's happening with search on these cities, expecting that Tel Aviv and Jerusalem would be found.

**Comment by missinglink** [comment]
heya,

There's two different issues going on here.

#### Tel Aviv

It seems that whosonfirst lists it as both a `region` and a `county`, you can get those records by removing the layers filter:

```
/v1/search?text=Tel Aviv
```

This doesn't sound correct, it might be worth opening an issue with the WOF team to ask why Tel Aviv is not currently classified as a `locality`.

---

#### Jerusalem

There is a difference in sorting between a query with a layers filter and one without:

```
/v1/search?layers=locality&text=Jerusalem
1)	Jerusalem, Australia
2)	Jerusalem, Palestine
3)	Jerusalem, OH, USA
4)	Jerusalém, Brazil
5)	Jerusalém, Brazil

/v1/search?text=Jerusalem
 1)	Jerusalem, Palestine
 2)	Jerusalem, Australia
 3)	Jerusalem, Israel
 4)	Jerusalem, Palestine
 5)	Jerusalem, Palestine
...
```

I'm not sure why this is happening, the sorting is based on population counts and place types (such as locality, localadmin etc) and so *should* sort the same. This is definitely a bug because the top two results in both queries are `locality`.

cc/ @trescube who can provide some more insight into why the ordering might be different depending on the layers filter.

**Comment by rmglennon** [comment]
Thanks for the review, @missinglink! I've put in https://github.com/whosonfirst-data/whosonfirst-data/issues/961 about the Tel Aviv locality/county/region question.

**Comment by trescube** [comment]
Yeah, Tel Aviv should probably be a locality.  A number of the things listed as localities of [Tel Aviv](https://whosonfirst.mapzen.com/spelunker/id/85672549/descendants/?exclude=nullisland) (the region) should probably be [neighbourhoods](https://en.wikipedia.org/wiki/Neighborhoods_of_Tel_Aviv).
